From 99589573cdbdcb0aa440794aae84e7f662c46aa3 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Tue, 26 Mar 2024 08:25:13 +0100
Subject: [PATCH] media: remove V4L2_CID_ISP_STAT_LOCATION ctrl definition

The DCMIPP driver doesn't need anymore the V4L2_CID_ISP_STAT_LOCATION
ctrl since a new mode to capture all location have been introduced
and fine control will be added using a profile ctrl.

Change-Id: Ib0a4637250939b66879e2926e253abc9f410c04e
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/369275
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/374630
---
 drivers/media/v4l2-core/v4l2-ctrls-defs.c | 1 -
 include/uapi/linux/v4l2-controls.h        | 6 ------
 2 files changed, 7 deletions(-)

diff --git a/drivers/media/v4l2-core/v4l2-ctrls-defs.c b/drivers/media/v4l2-core/v4l2-ctrls-defs.c
index ea44c14eadf6..e9e97064381e 100644
--- a/drivers/media/v4l2-core/v4l2-ctrls-defs.c
+++ b/drivers/media/v4l2-core/v4l2-ctrls-defs.c
@@ -1164,7 +1164,6 @@ const char *v4l2_ctrl_get_name(u32 id)
 	case V4L2_CID_DEINTERLACING_MODE:	return "Deinterlacing Mode";
 	case V4L2_CID_DIGITAL_GAIN:		return "Digital Gain";
 	case V4L2_CID_ISP_STAT_REGION:		return "ISP Stat Region";
-	case V4L2_CID_ISP_STAT_LOCATION:	return "ISP Stat Location";
 	case V4L2_CID_ISP_STAT_AVG_FILTER:	return "ISP Stat Average Filter";
 	case V4L2_CID_ISP_STAT_BIN_COMP:	return "ISP Stat Bin Component";
 
diff --git a/include/uapi/linux/v4l2-controls.h b/include/uapi/linux/v4l2-controls.h
index a36316bb4471..2d81b73a6434 100644
--- a/include/uapi/linux/v4l2-controls.h
+++ b/include/uapi/linux/v4l2-controls.h
@@ -1230,12 +1230,6 @@ struct v4l2_ctrl_isp_stat_region {
 	__u32 height[25];
 };
 
-#define V4L2_CID_ISP_STAT_LOCATION		(V4L2_CID_IMAGE_PROC_CLASS_BASE + 11)
-enum v4l2_isp_stat_location {
-	V4L2_STAT_LOCATION_BEFORE_PROC	= 0,
-	V4L2_STAT_LOCATION_AFTER_DEMO	= 1,
-};
-
 #define V4L2_CID_ISP_STAT_AVG_FILTER		(V4L2_CID_IMAGE_PROC_CLASS_BASE + 12)
 enum v4l2_isp_stat_avg_filter {
 	V4L2_STAT_AVG_FILTER_NONE	= 0,
-- 
2.39.5

